### ISSUE TITLE
Usefulness of distinct 'single' and 'multiple' styles for sort-imports

### ISSUE THREAD

**Issue by kj** [eslint/eslint#13655]
**What rule do you want to change?**

sort-imports

**The issue**

Currently, the 'single' style from the 'memberSyntaxSortOrder' option, covers both of these lines:

```
import a from 'a';
import {b} from 'b';
```

This means that this:

```
import {a} from 'a';
import {b, c} from 'b';
```

Would be incorrect, as the first line is considered 'single', while the second line is considered 'multiple', yet clearly they're the same import style and all that differs is the number of imported members. I don't see much value in this. I expected that 'single' imports would mean 'default imports', i.e. the first line in the first example, but not the second line, which I would consider 'multiple' (even though it is only importing one member) as it is a different style. However, it is also possible to import the default export and other members within the same line, e.g.

```js
import React, {useEffect, useState} from 'react';
```

Perhaps this is the reasoning for the distinction between single and multiple, rather than between default and non-default. However, I don't see the value in treating the two import lines in the first example as the same style (clearly they aren't), and grouping on the number of imported members (which does not feel much like a 'style'). I feel that it would be better to either remove the distinction between single and multiple, or change the groups/styles to something like 'none', 'all', withDefault', and 'withoutDefault', so that you'd end up with something like this:

```js
import 'none';
import * from 'all1';
import * from 'all2';
import a from 'a'; // has a default import
import b, {f, g} from 'b'; // has a default import
import {d} from 'd'; // does not have a default import
import {e} from 'e'; // does not have a default import
```

**Are you willing to submit a pull request to implement this change?**

Possibly.

**Comment by anikethsaha** [comment]
Thanks for the issue. 

I would prefer to have a name like `default` in order to differ from single. 

Previous discussion : https://github.com/eslint/eslint/issues/12065 , that was not accepted. 
But I can we can re-consider that one here again as there is a room for refining the options again. 

IMO, I would add one more `default` name.


Also, I doubt this might be a stylistic rule category, if this is so, then this will fall under our new [policy](https://eslint.org/blog/2020/05/changes-to-rules-policies)

**Comment by mdjermanovic** [comment]
This is indeed a stylistic rule, which falls under our new policy.

However, I would argue that this is a design bug. The option is named "member _syntax_ sort order", but it doesn't make a distinction between `import a` and `import { a }`.

I support a backward compatible change, though I'd personally prefer the design from #12065. Anyway, let's see first if there's interest from the team to implement this change.

**Comment by kj** [comment]
Thank you for your positive reception, I appreciate that regardless of whether this goes anywhere! The proposal in #12065  (none, all, default, mixed, named) does make more sense to me, being syntax-based. A backward-compatible change then would be to treat that as an alternate set of options, as the previous issue suggests, rather than modifying the existing set?

**Comment by mdjermanovic** [comment]
> A backward-compatible change then would be to treat that as an alternate set of options, as the previous issue suggests, rather than modifying the existing set?

Exactly! I think we shouldn't even deprecate the existing set of items (`["none", "all", "multiple", "single"]`), but just allow the alternate set instead, so the users could choose either based on their preferences.
